test/integration/restart_task_test.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+require 'test_helper'
+require 'kubernetes-deploy/restart_task'
+
+class RestartTaskTest < KubernetesDeploy::IntegrationTest
```

Comment:
Good test coverage! Suggestions for two more tests:

* `wait_for_rollout` fails to see the containers come back up (perhaps test this by making them sleep for too long?)
* Restarting after having already restarted, i.e. `RESTARTED_AT` is already set.